src/System.Management.Automation/utils/Verbs.cs

```diff
@@ -1311,6 +1309,68 @@ static Verbs()
 #endif
         }
 
+        private static Type[] VerbTypes => new Type[] {
+            typeof(VerbsCommon),
+            typeof(VerbsCommunications),
+            typeof(VerbsData),
+            typeof(VerbsDiagnostic),
+            typeof(VerbsLifecycle),
+            typeof(VerbsOther),
+            typeof(VerbsSecurity)
+        };
+
+        private static string GetVerbGroupDisplayName(Type verbType) => verbType.Name.Substring(5);
+
+        internal static IEnumerable<VerbInfo> GetVerbsByGroup(string[] verbs, string[] groups)
+        {
+            Collection<WildcardPattern> verbPattern = SessionStateUtilities.CreateWildcardsFromStrings(
+                verbs ?? new string[] { "*" },
+                WildcardOptions.IgnoreCase);
+
+            foreach (Type verbType in VerbTypes) 
+            {
+                string verbGroupName = GetVerbGroupDisplayName(verbType);
+
+                bool hasVerbGroup = SessionStateUtilities.CollectionContainsValue(
+                    groups ?? Array.Empty<string>(),
+                    verbGroupName,
+                    StringComparer.OrdinalIgnoreCase);
+
+                if (groups is null || hasVerbGroup)
```

Comment:
Do we really need the `groups is null` check guess hasVerbGroup is false if group is null.